internal/host/plugin/repository_host_catalog.go

```diff
@@ -143,6 +146,279 @@ func (r *Repository) CreateCatalog(ctx context.Context, c *HostCatalog, _ ...Opt
+	}
+
+	plgResp, err := plgClient.OnUpdateCatalog(ctx, &plgpb.OnUpdateCatalogRequest{
+		CurrentCatalog: currPlgHc,
+		NewCatalog:     newPlgHc,
+		Persisted:      currentCatalogPersisted,
+	})
+	if err != nil {
+		if status.Code(err) != codes.Unimplemented {
+			return nil, db.NoRowsAffected, db.NoRowsAffected, errors.Wrap(ctx, err, op)
+		}
+	}
+
+	// Success for OnUpdateCatalog. This means we can start updating.
+	// First, check for returned persisted data and encrypt it.
+	var hcSecret *HostCatalogSecret
+	var deleteSecrets bool
+	if plgResp != nil && plgResp.GetPersisted().GetSecrets() != nil {
+		if len(plgResp.GetPersisted().GetSecrets().GetFields()) == 0 {
+			// Flag the secret to be deleted.
+			hcSecret, err = newHostCatalogSecret(ctx, currentCatalog.GetPublicId(), plgResp.GetPersisted().GetSecrets())
+			if err != nil {
+				return nil, db.NoRowsAffected, db.NoRowsAffected, errors.Wrap(ctx, err, op)
+			}
+
+			deleteSecrets = true
+		} else {
+			hcSecret, err = newHostCatalogSecret(ctx, currentCatalog.GetPublicId(), plgResp.GetPersisted().GetSecrets())
+			if err != nil {
+				return nil, db.NoRowsAffected, db.NoRowsAffected, errors.Wrap(ctx, err, op)
+			}
+			dbWrapper, err := r.kms.GetWrapper(ctx, c.ScopeId, kms.KeyPurposeDatabase)
+			if err != nil {
+				return nil, db.NoRowsAffected, db.NoRowsAffected, errors.Wrap(ctx, err, op, errors.WithMsg("unable to get db wrapper"))
+			}
+			if err := hcSecret.encrypt(ctx, dbWrapper); err != nil {
+				return nil, db.NoRowsAffected, db.NoRowsAffected, errors.Wrap(ctx, err, op)
+			}
+		}
+	}
+
+	// Get the oplog.
+	oplogWrapper, err := r.kms.GetWrapper(ctx, c.ScopeId, kms.KeyPurposeOplog)
+	if err != nil {
+		return nil, db.NoRowsAffected, db.NoRowsAffected, errors.Wrap(ctx, err, op, errors.WithMsg("unable to get oplog wrapper"))
+	}
+
+	var catalogsUpdated, secretsUpdated int
+	var returnedCatalog *HostCatalog
+	_, err = r.writer.DoTx(
+		ctx,
+		db.StdRetryCnt,
+		db.ExpBackoff{},
+		func(_ db.Reader, w db.Writer) error {
+			msgs := make([]*oplog.Message, 0, 3)
+			ticket, err := w.GetTicket(c)
+			if err != nil {
+				return errors.Wrap(ctx, err, op, errors.WithMsg("unable to get ticket"))
+			}
+
+			if len(dbMask) != 0 || len(nullFields) != 0 {
+				returnedCatalog = c.clone()
+				var cOplogMsg oplog.Message
+				catalogsUpdated, err = w.Update(
+					ctx,
+					returnedCatalog,
+					dbMask,
+					nullFields,
+					db.NewOplogMsg(&cOplogMsg),
+					db.WithVersion(&version),
+				)
+				if err != nil {
+					return errors.Wrap(ctx, err, op)
+				}
+				if catalogsUpdated > 1 {
+					return errors.New(ctx, errors.MultipleRecords, op, "more than 1 resource would have been updated")
+				}
+				msgs = append(msgs, &cOplogMsg)
+			} else {
+				// Returned catalog needs to be the old copy, as no fields in the
+				// catalog itself are being updated (note: secrets may still be
+				// updated).
+				returnedCatalog = currentCatalog.clone()
+			}
+
+			if hcSecret != nil {
```

Comment:
The catalog's version should be incremented if it's secret is updated even if it is the only thing that got updated.

Here is an example of how oidc auth methods increase the version when the primary table isn't the thing being updated: https://github.com/hashicorp/boundary/blob/main/internal/auth/oidc/repository_auth_method_update.go#L269

Reply:
Done!